src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -542,9 +542,6 @@ public ActivityProfileState isActive( final AlignmentContext context, final Refe
         final AssemblyResultSet untrimmedAssemblyResult =  AssemblyBasedCallerUtils.assembleReads(region, givenAlleles, hcArgs, readsHeader, samplesList, logger, referenceReader, assemblyEngine, aligner, !hcArgs.doNotCorrectOverlappingBaseQualities);
         
         final SortedSet<VariantContext> allVariationEvents = untrimmedAssemblyResult.getVariationEvents(hcArgs.maxMnpDistance);
-        // TODO - line bellow might be unnecessary : it might be that assemblyResult will always have those alleles anyway
-        // TODO - so check and remove if that is the case:
```

Comment:
It feels so good to finally take care of that TODO that's been around for longer than I have.